src/foam/nanos/auth/ChangePasswordView.js

```diff
@@ -9,267 +9,170 @@ foam.CLASS({
-    'auth',
-    'notify',
-    'resetPasswordToken',
     'stack',
     'theme',
     'user'
   ],
 
   requires: [
-    'foam.nanos.auth.User',
-    'foam.u2.detail.SectionView'
+    'foam.u2.detail.SectionView',
+    'foam.nanos.auth.SendPassword',
+    'foam.nanos.auth.UpdatePassword',
+    'foam.nanos.auth.ResetPassword'
   ],
 
   css: `
     ^ {
       background: #ffffff;
       height: 100vh;
     }
-    ^ .centerVertical {
-      max-width: 40vw;
-      margin: 6% auto;
-    }
-    ^ .logoCenterVertical {
-      margin: 0 auto;
-      text-align: center;
-      display: block;
-    }
-    ^ .horizontal {
-      padding: 0 0 1vh 2vh;
-      max-width: 98%;
-    }
-    ^ .logoHorizontal {
-      padding-left: 2vh;
-    }
-    ^ .top-bar {
+    ^top-bar {
       background: /*%LOGOBACKGROUNDCOLOUR%*/ #202341;
       width: 100%;
       height: 8vh;
       border-bottom: solid 1px #e2e2e3;
     }
-    ^ .top-bar img {
+    ^top-bar img {
       height: 4vh;
       padding-top: 2vh;
       display: block;
       margin: 0 auto;
     }
-    ^ .title-top {
-      font-size: 2.5em;
-      padding-top: 2vh;
-      font-weight: bold;
-      text-align: center;
-      margin-bottom: 0
+    ^content {
+      max-width: 30vw;
+      margin: 4vh auto 0 auto;
+    }
+    ^section {
+    }
+    /* title  */
+    ^ ^section h2 {
+      margin-top: 0;
+      margin-bottom: 4vh;
+      font-size: 2.5rem;
     }
-    ^ .subtitle {
-      color: #525455;
-      font-size: 1em;
+    /* subtitle */
+    ^ ^section .subtitle {
+      color: /*%GREY2%*/ #8e9090;
+      font-size: 1rem;
+      margin-top: 0;
+      margin-bottom: 3vh;
       line-height: 1.5;
-      margin-bottom: 2vh;
       text-align: center;
     }
-    ^ .contents {
-      max-width: 25vw;
-      margin: 0 auto;
-    }
-    ^ .submitBtn {
+    ^link {
+      color: /*%PRIMARY3%*/ #604aff;
+      cursor: pointer;
       text-align: center;
+      padding-top: 1.5vh;
     }
   `,
 
-  sections: [
-    {
-      name: 'resetPasswordSection',
-      title: '',
-      help: 'Create a new password for your account',
-    }
-  ],
-
   properties: [
     {
       class: 'Boolean',
-      name: 'horizontal',
-      documentation: `This property toggles the view from a centered vertical view to a
-      horizontally property display view.`,
-      hidden: true
-    },
-    {
-      class: 'Boolean',
-      name: 'topBarShow',
+      name: 'showHeader',
       documentation: `This property toggles the view from having a top bar displayed.`,
       hidden: true,
-      value: true
-    },
-    {
-      class: 'String',
-      name: 'token',
-      documentation: `This property toggles the view from updating a user password to resetting a user password.`,
       factory: function() {
-        var searchParams = new URLSearchParams(location.search);
-        return searchParams.get('token');
-      },
-      hidden: true
+        // implement a logic to hide logo in certain places
+        return true;
```

Comment:
Is this in the factory? Couldn't you just use value:?

Reply:
Yeah, I could. I made a factory so that when we actually want to add implementation to hide logo, we can just write the logic here.

Reply:
Should I change it to value?